eng/pipelines/backup_build_artifacts.yml

```diff
@@ -0,0 +1,102 @@
+        displayName: 'Download product RTM'
+        continueOnError: true # in case build didn't publish this artifact
+        inputs:
+          buildType: specific
+          project: $(resources.pipeline.officialbuild.projectID)
+          definition: $(resources.pipeline.officialbuild.pipelineID)
+          artifactName: 'VS15-RTM'
+          targetPath: $(Build.ArtifactStagingDirectory)/backup/VS15-RTM
+
+      - task: DownloadPipelineArtifact@2
+        displayName: 'Download nupkgs'
+        continueOnError: true # in case build didn't publish this artifact
+        inputs:
+          buildType: specific
+          project: $(resources.pipeline.officialbuild.projectID)
+          definition: $(resources.pipeline.officialbuild.pipelineID)
+          artifactName: 'nupkgs - NonRTM'
+          targetPath: $(Build.ArtifactStagingDirectory)/backup/VS15/nupkgs
+
+      - task: DownloadPipelineArtifact@2
+        displayName: 'Download nupkgs RTM'
+        continueOnError: true # in case build didn't publish this artifact
+        inputs:
+          buildType: specific
+          project: $(resources.pipeline.officialbuild.projectID)
+          definition: $(resources.pipeline.officialbuild.pipelineID)
+          artifactName: 'nupkgs - RTM'
+          targetPath: $(Build.ArtifactStagingDirectory)/backup/VS15-RTM/nupkgs
+
+      - task: DownloadPipelineArtifact@2
+        displayName: 'Download symbols NonRTM'
+        continueOnError: true # in case build didn't publish this artifact
+        inputs:
+          buildType: specific
+          project: $(resources.pipeline.officialbuild.projectID)
+          definition: $(resources.pipeline.officialbuild.pipelineID)
+          artifactName: 'symbols - NonRTM'
+          targetPath: $(Build.ArtifactStagingDirectory)/backup/symbols
+
+      - task: DownloadPipelineArtifact@2
+        displayName: 'Download symbols RTM'
+        continueOnError: true # in case build didn't publish this artifact
+        inputs:
+          buildType: specific
+          project: $(resources.pipeline.officialbuild.projectID)
+          definition: $(resources.pipeline.officialbuild.pipelineID)
+          artifactName: 'symbols - RTM'
+          targetPath: $(Build.ArtifactStagingDirectory)/backup/symbols-rtm
+
+      - pwsh: gci -recurse -file | Resolve-Path -Relative
+        workingDirectory: $(Build.ArtifactStagingDirectory)/backup
+        displayName: "List files being backed up"
+
+      - task: artifactDropTask@0
+        displayName: "Upload Drop"
+        inputs:
+          dropServiceURI: 'https://devdiv.artifacts.visualstudio.com'
+          buildNumber: 'Backup/$(System.TeamProject)/NuGet.Client/$(resources.pipeline.officialbuild.pipelineName)/$(resources.pipeline.officialbuild.runName)'
+          sourcePath: "$(Build.ArtifactStagingDirectory)/backup"
+          toLowerCase: false
+          usePat: true
+          dropMetadataContainerName: "DropMetadata-Backup"
+          retentionDays: 186
```

Comment:
There should be organization-level decisions made on how to store shipped assets.
You've set this to about 6 months, currently. After 6 months, we'll lose the assets, and I'd think that'd be a problem. 
Do we need to ask someone in the division about this?

Reply:
Isn't that 6 months for binaries that did not ship?

Binaries we ship are retained forever.

Reply:
Right, as I wrote in my comment, one option is to:

> have our release pipeline retain this drop "permanently", but only for builds that we actually ship

The VS insertion in our release pipeline already retains the `Product/NuGet/` drop permanently. We could add another step in the release pipeline to retain this `Backup/NuGet/` drop permanently as well, but let builds that were never inserted be deleted after 6 months. Since the release pipeline isn't yaml, I couldn't modify it as part of this PR.

There are also questions if we should retain all builds that were inserted, even those that were overwritten by another insertion before it ships to customers?  Or should we have a manual step in the release pipeline that is triggered as part of our release process, just like publishing packages to nuget.org?

If anyone can think of a good reason to keep all our official builds permanently, we can do that as well, but as I wrote in my comment, the org is trying to reduce unnecessary costs. So builds that are never even inserted, they seem pointless to retain "forever" (until manually deleted)